sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationPolicyTests.java

```diff
@@ -0,0 +1,285 @@
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class AttestationPolicyTests extends AttestationClientTestBase {
+    private static final String DISPLAY_NAME_WITH_ARGUMENTS = "{displayName} with [{arguments}]";
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getPolicyClients")
+    void testGetAttestationPolicy(HttpClient client, String clientUri, AttestationType attestationType)
+        throws ParseException {
+
+        AttestationClientBuilder attestationBuilder = getBuilder(client, clientUri);
+
+        PolicyResponse policyResponse = attestationBuilder.buildPolicyClient().get(attestationType);
+
+        verifyBasicGetAttestationPolicyResponse(client, clientUri, attestationType, policyResponse);
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getPolicyClients")
+    void testGetAttestationPolicyAsync(HttpClient client, String clientUri, AttestationType attestationType) {
+        AttestationClientBuilder attestationBuilder = getBuilder(client, clientUri);
+
+        StepVerifier.create(attestationBuilder.buildPolicyAsyncClient().get(attestationType))
+            .assertNext(response -> assertDoesNotThrow(() -> verifyBasicGetAttestationPolicyResponse(client, clientUri, attestationType, response))).verifyComplete();
+    }
+
+    /**
+     * Verifies attestation policy set operations.
+     *
+     * This method iterates over the attestation types and instances and attempts to set attestation policies on
+     * each instance.
+     *
+     * For AAD instances, we try set policy with two separate policies, one secured and the other unsecured.
+     * For Isolated instances, we try to set policy with a policy document signed by the isolated signer.
+     *
+     * After the policy is set, the test method cleans up by resetting attestation policy to the default policy.
+     *
+     * @param httpClient HTTP Client used for operations.
+     * @param clientUri Base URI for attestation instance.
+     * @param attestationType AttestationType on which to set policy.
+     * @throws JOSEException Should never be thrown.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getPolicyClients")
+    void testSetAttestationPolicy(HttpClient httpClient, String clientUri, AttestationType attestationType)
+        throws JOSEException, InvalidKeySpecException, NoSuchAlgorithmException, ParseException {
+
+        ClientTypes clientType = classifyClient(clientUri);
+        // We can't set attestation policy on the shared client, so just exit early.
+        if (clientType.equals(ClientTypes.SHARED)) {
```

Comment:
Use `assumeTrue` for instances where the test cannot be executed in that instance.